src/libs/elektra/keyset.c

```diff
@@ -709,6 +709,38 @@ ssize_t ksSearchInternal (const KeySet * ks, const Key * toAppend)
 	}
 }
 
+/**
+ * Efficient search in a key set, either yielding the actual index
+ * of the key, if the ley has been found within the key set, or a
+ * negative value indicating the insertion index of the key, if the
+ * key would be inserted.
+ *
+ * @code
+
+ssize_t result = ksSearch(ks, key);
+
+if (result >= 0)
+{
+	ssize_t position = result;
+	// Seems like the key already exist.
+} else {
+	ssize_t insertpos = -result-1;
+	// Seems like the key does not exist.
+}
+ * @endcode
+ *
+ * @param ks the keyset to work with
+ * @param key the key to check
+ * @return position where the key is (>=0) if the key was found
+ * @return -insertpos -1 (< 0) if the key was not found
+ *    so to get the insertpos simple do: -insertpos -1
+ */
```

Comment:
We also need `@see` to ksLookup.

@lawli3t can you please also do a review of this docu here.

Reply:
Updated ksSearch docs with `@see ksLookup() for retrieving the found key`